pyflakes/checker.py

```diff
@@ -147,6 +147,17 @@ def redefines(self, other):
         return isinstance(other, Definition) and self.name == other.name
 
 
+class StarImportation(Importation):
+    """A binding created by an 'from x import *' statement."""
+
+    def __init__(self, name, source):
+        super(StarImportation, self).__init__('*', source)
+        # Each star importation needs a unique name, and
+        # may not be the module name otherwise it will be deemed imported
+        self.name = name + '.*'
```

Comment:
I've improved this changeset considerably, creating a separate StarImportation binding for each import.  This line here is a new 'ugly'.  I havent thought of a better way yet.